molecule/config.py

```diff
@@ -47,10 +47,12 @@ def populate_instance_names(self, platform):
         :param platform: platform name to pass to ``format_instance_name`` call
         :return: None
         """
-        for instance in self.config['vagrant']['instances']:
-            instance['vm_name'] = utilities.format_instance_name(
-                instance['name'], platform,
-                self.config['vagrant']['instances'])
+
+        if 'vagrant' in self.config:
```

Comment:
Wouldn't ansible.cfg only be rewritten if we run `molecule init`.  Other than that it should never be re-generated.